### PR TITLE
Extend the pex tool to support debuggers

### PR DESCRIPTION
**Pull request by tiagovtristao**
This PR mainly extends the pex tool to support debuggers. Some refactoring, including files being moved around, was done for better clarity.

### CODE REVIEW

File: rules/python_rules.build_defs [hunk 1]
```diff
@@ -103,6 +103,9 @@ def python_binary(name:str, main:str, srcs:list=[], resources:list=[], out:str=N
     by the dependent python_library rules, and this rule simply builds the
     metadata for it and concatenates them all together.
 
+    This target can be debugged via:
+        plz debug [--debugger=[gdb|debugpy]] [--port] //:target
```
Comment (Tatskaari) [inline rules/python_rules.build_defs:107]: Do you mean pdb?

File: rules/python_rules.build_defs [hunk 2]
```diff
@@ -145,6 +148,12 @@ def python_binary(name:str, main:str, srcs:list=[], resources:list=[], out:str=N
         test_only=test_only,
     )
 
+    debug_cmd, debugger = _debug_cmd("./$OUT")
+    if debug_cmd:
+        if debugger == "debugpy":
+            cmd = cmd.replace(' --zip_safe', '')
```
Comment (Tatskaari) [inline rules/python_rules.build_defs:154]: qq: why is this needed?

File: rules/python_rules.build_defs [hunk 3]
```diff
@@ -205,6 +215,9 @@ def python_test(name:str, srcs:list, data:list|dict=[], resources:list=[], deps:
     on which is set for the test runner, which can be configured either via the python_test_runner
     package property or python.testrunner in the config.
 
+    This target can be debugged via:
+        plz debug [--debugger=[gdb|debugpy]] [--port] //:target
```
Comment (Tatskaari) [inline rules/python_rules.build_defs:219]: Same here

**Review by Tatskaari — APPROVED** [review]
Looks good to me! Some of this will need to be migrated over to the plugin though.

File: rules/python_rules.build_defs (hunk 1 above)
Comment (tiagovtristao) [reply rules/python_rules.build_defs:107]: Good catch

File: rules/python_rules.build_defs (hunk 2 above)
Comment (tiagovtristao) [reply rules/python_rules.build_defs:154]: debugpy was freaking during some file access of some some sort IIRC
